mne/viz/evoked.py

```diff
@@ -1529,162 +1689,204 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     tmin, tmax = times[0], times[-1]
-    if (tmin >= 0 or tmax <= 0) and vlines == [0.]:
-        vlines = list()
+
+    if vlines is "auto" and (tmin < 0 and tmax > 0):
+        vlines = [0.]
+    assert isinstance(vlines, list)
 
     if isinstance(picks, Integral):
         picks = [picks]
-    elif len(picks) == 0:
-        warn("No picks, plotting the GFP ...")
+    elif picks is None:
+        logger.info("No picks, plotting the GFP ...")
         gfp = True
-        picks = _pick_data_channels(example.info)
+        picks = _pick_data_channels(info)
+
+    if not isinstance(picks, (list, np.ndarray)):
+        raise TypeError("picks should be a list or np.array of integers. "
+                        "Got %s." % type(picks))
 
-        if len(picks) == 0:
-            raise ValueError("No valid channels were found to plot the GFP. " +
-                             "Use 'picks' instead to select them manually.")
+    if len(picks) == 0:
+        raise ValueError("No valid channels were found to plot the GFP. " +
+                         "Use 'picks' instead to select them manually.")
 
     if ylim is None:
         ylim = dict()
 
     # deal with picks: infer indices and names
     if gfp is True:
-        show_sensors = False if show_sensors is None else show_sensors
+        if show_sensors is None:
+            show_sensors = False  # don't show sensors for GFP
         ch_names = ['Global Field Power']
         if len(picks) < 2:
             raise ValueError("A GFP with less than 2 channels doesn't work, "
-                             "please pick more channels.")
+                             "please pick more than %d channels." % len(picks))
+    else:
+        if show_sensors is None:
+            show_sensors = True  # show sensors when not doing GFP
+        ch_names = [one_evoked.ch_names[pick] for pick in picks]
+
+    picks_by_types = channel_indices_by_type(info, picks)
+    # keep only channel types for which there is a channel:
+    ch_types = [t for t in picks_by_types if len(picks_by_types[t]) > 0]
+
+    # let's take care of axis and figs
+    if axes is not None:
+        if not isinstance(axes, list):
+            axes = [axes]
+        from .utils import _validate_if_list_of_axes
+        _validate_if_list_of_axes(axes, obligatory_len=len(ch_types))
     else:
-        if not isinstance(picks[0], (int, np.integer)):
-            msg = "'picks' must be int or a list of int, not {0}."
-            raise ValueError(msg.format(type(picks)))
-        show_sensors = True if show_sensors is None else show_sensors
-        ch_names = [example.ch_names[pick] for pick in picks]
-    ch_types = list(set(channel_type(example.info, pick_)
-                    for pick_ in picks))
-    # XXX: could possibly be refactored; plot_joint is doing a similar thing
-    if any([type_ not in _VALID_CHANNEL_TYPES for type_ in ch_types]):
-        raise ValueError("Non-data channel picked.")
+        axes = []
```

Comment:
Could do instead:

`axes = [plt.subplots(figsize=(8, 6))[1] for _ in range(len(ch_types))]`